spring-cloud-context/src/main/java/org/springframework/cloud/bootstrap/config/PropertySourceBootstrapConfiguration.java

```diff
@@ -154,36 +164,39 @@ private void setLogLevels(ConfigurableApplicationContext applicationContext,
 	}
 
 	private void insertPropertySources(MutablePropertySources propertySources,
-			CompositePropertySource composite) {
+			List<PropertySource<?>> composite) {
 		MutablePropertySources incoming = new MutablePropertySources();
-		incoming.addFirst(composite);
+		List<PropertySource<?>> reversedComposite = new ArrayList(composite);
+		// Reverse the list so that when we call addFirst below we are maintaining the
+		// same order of PropertySournces
+		// Wherever we call addLast we can use the order in the List since the first item
+		// will end up before the rest
+		Collections.reverse(reversedComposite);
+		reversedComposite.stream().forEach(p -> incoming.addFirst(p));
```

Comment:
This line seems out of place to me.

Reply:
I could move `Collections.reverse(reversedComposite);` to the line above so it is a bit clearer.  
```
List<PropertySource<?>> reversedComposite = Collections.reverse(new ArrayList(composite));
```

As you know, the property source list matters....

Consider we have a `composite` of PropertySource1, PropertySource2, and PropertySource3.  If I just iterated through the list and called `incoming.addFirst(p)` the resulting `incoming` property source list would have PropertySource3, PropertySource2, PropertySource1, essentially the reverse order.  So I reverse the list in order to make sure the property sources in `composite` have the same order.

Reply:
:+1: